common/user/info.go

```diff
@@ -0,0 +1,32 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package user
+
+// Info holds information about user
+// This information is used to generate corresponding JWT claim
+type Info struct {
+	Name     string
+	Password string
+	Admin    bool
+	Groups   []string
```

Comment:
What is group?

Reply:
In Cadence we have OAuth authorization already implemented. This struct mainly reflects model for authorization already in use.

On server side, you can update domain metadata with keys: READ_GROUPS and WRITE_GROUPS to define what "group" has what permissions. Then groups and admin flag are checked in JWT claim verification part when "access controlled handler" is used.

References to get more context: 
* [JWT Claim struct](https://github.com/uber/cadence/blob/5f9eb08abfdc1f5ec68b097a5e5e93a6fcdf283c/common/authorization/oauthAuthorizer.go#L48)
* [JWT claim verification](https://github.com/uber/cadence/blob/5f9eb08abfdc1f5ec68b097a5e5e93a6fcdf283c/common/authorization/oauthAuthorizer.go#L148)
* [Access controlled handler authorize flow example](https://github.com/uber/cadence/blob/5f9eb08abfdc1f5ec68b097a5e5e93a6fcdf283c/service/frontend/accessControlledHandler.go#L747)